### ISSUE TITLE
Sanity checks missing

### ISSUE THREAD

**Issue by johannesbottcher** [derric/cleanthesis#39]
The package doesn't do any checking at all? 

```syntax=latex
\documentclass{memoir}
\usepackage{cleanthesis}
\begin{document}
\end{document}
```

What is your definition of a template? c.f. [template Confusion](https://github.com/johannesbottcher/templateConfusion)

**Comment by derric** [comment]
@johannesbottcher: For me, a template is an empty (lacking content) thing. It allows you to create a document with pre-defined properties, but with your own content. The document creator does not work on the presentation, but on the content. As long as the document author can live with the pre-defined document properties (visuals, such as colors, spacing, fonts and sizes, margins, icons, footers, ...), chances are high that no technical problems will occur.

Therefore, I argue to not use the cleanthesis.sty only (just as in the example above). The template consists of more files. In fact, all the placeholder files allow the document creator to create his/her thesis by replacing the example content all files with own content. But, to be honest, this is an issue in the current master, since the MANIFEST.md as well as the CTAN package does not mention/include all the my-example files. But it definitively should.

**Comment by derric** [comment]
@johannesbottcher: Furthermore, a template can also allow the user (document creator) to adjust the template itself by using options, package parameters, other documented and explained commands/scripts. But everything else, other use cases might result in problems, which usually need some expertise in order to work around.

**Comment by johannesbottcher** [comment]
@derric  I see we mainly agree on tempaltes :-)

In my point of view,  a LateX package should be usable standalone, or check that the things it needs are there (and abort if not). But that really is a complicated issue at the end.

**Comment by derric** [comment]
@johannesbottcher: Furthermore ;) CleanThesis is good example the next point. I developed the template for my own Diplomarbeit - only for my own. However, and this probably is the most frequent motivation of template authors, I thought it would be good idea to allow others the save time by not having to think about visuals, but focus on their content. And I still think it was right. At this point, I do not agree with your comment at http://tex.stackexchange.com/questions/306382/header-and-text-font-isnt-changing:
> [...] So a little advice to all wanting to write up a nice template: Don't.

I think, the challenge is more about the communication between the template author and the users (document creators). The template author needs to communicate the limits. And the user has two options:
1. Stick to the template. Only use features/functions that are provided by the template. Regarding technical problems, this should be the safe, painless way.
2. User thinks: "Oh. This is good starting point, but I want this to be like that, this should looks different, and actually, I want to use it with this". In this case, the user should expect some technical challenges or problems. The user should accept that this way needs an extra investment of time for non-content related tasks.

**Comment by derric** [comment]
Yes, correct. I totally agree that in case of the CleanThesis template, there should user feedback if something is missing or even used in some strange way. Solving the latter is, in my opinion, not possible. The only thing a template author can do is to communicate the correct and typical use case.

**Comment by johannesbottcher** [comment]
That communication thing you mention is a very very good point. The style/look/visual appearance is what most templates define and changing that is the hard part.Making that clear right from the start would be a good thing.  
http://chat.stackexchange.com/transcript/message/30663861#30663861
